src/commands/project/__tests__/list.test.js

```diff
@@ -0,0 +1,133 @@
+    before(function () {
+      this.invokeProject = require('../../project').invoke
+
+      this.invoke = (argv, notify, options) => this.invokeProject(['list', ...argv], notify, options)
+      this.notify = msg => {
+        this.notifications.push(msg)
+      }
+      this.formatError = msg => `__FMT: ${msg}`
+      this.lgJWT = 'not.a.real.token'
+      this.lgUser = {id: 'not.a.real.id', roles: ['player']}
+    })
+    beforeEach(function () {
+      this.notifications = []
+    })
+
+    afterEach(function () {
+      nock.cleanAll()
+    })
+
+    it('notifies with the usage message when requested', notifiesWithUsageMessageForDashH)
+    it('notifies with a warning if too many positional arguments are provided', notifiesWithUsageHintForInvalidArgs(['these', 'are', 'arguments']))
+    it('notifies with an error message when invoked by a non-player', function () {
+      return notifiesWithErrorIfNotAPlayer([]).bind(this)()
+    })
+    it('notifies of API invocation errors', notifiesWithErrorForAPIErrors([]))
+    it('notifies of GraphQL invocation errors', notifiesWithErrorForGraphQLErrors([]))
+
+    const _summaryData = {
+      data: {
+        getProjectSummaryForPlayer: {
+          numActiveProjectsForCycle: 4,
+          numTotalProjectsForPlayer: 2,
+        }
+      }
+    }
+
+    const _projectList = {
+      data: {
+        getProjectsWithReviewResponsesForPlayer: [{
+          name: 'frightened-grouse',
+          artifactURL: 'http://example.com',
+          projectReviewResponses: [{
+            name: 'quality',
+            value: '88',
+          }, {
+            name: 'completeness',
+            value: '23',
+          }],
+        }, {
+          name: 'helpless-rat',
+          artifactURL: null,
+          projectReviewResponses: [{
+            name: 'quality',
+            value: null,
+          }, {
+            name: 'completeness',
+            value: null,
+          }],
+        }, {
+          name: 'magnificent-galah',
+          artifactURL: null,
+          projectReviewResponses: [{
+            name: 'quality',
+            value: null,
+          }, {
+            name: 'completeness',
+            value: null,
+          }],
+        }, {
+          name: 'ordinary-wapiti',
+          artifactURL: null,
+          projectReviewResponses: [{
+            name: 'quality',
+            value: null,
+          }, {
+            name: 'completeness',
+            value: null,
+          }],
+        }]
+      }
+    }
+
+    it('displays summary information when no options are passed', function (done) {
+      nock('http://game.learnersguild.test')
+        .post('/graphql')
+        .reply(200, _summaryData)
+
+      const {lgJWT, lgUser} = this
+      return this.invoke([], this.notify, {lgJWT, lgUser})
+        .then(() => {
+          expect(this.notifications[0]).to.match(/4 active projects.*participated in 2 projects/)
+          done()
+        })
+        .catch(err => done(err))
```

Comment:
Can't you just return the promise and not explicitly invoke the callback? Seeing this makes me worry I'm doing it wrong everywhere else.